### ISSUE TITLE
In IE8, HTML5 tags such as `<section>` aren't supported

### ISSUE THREAD

**Issue by sophiebits** [facebook/react#1030]
tl;dr: In IE8,

```
React.renderComponent(React.DOM.section(null, 'hi!'), document.body);
```

doesn't work:

![image](https://f.cloud.github.com/assets/6820/2095055/da7f81b8-8ed9-11e3-958f-cc46b659348d.png)

---

Background: To make HTML5 tags such as `<section>` work in IE8, one needs to do something along the lines of:

``` js
<script>
(function() {
var html5elements = ["header", "footer", "nav", "article", "section", "menu"];
for (var i = 0; i < html5elements.length; i++) {
    document.createElement(html5elements[i]);
}
})();
</script>
```

as well as create a CSS rule for them with `display: block;` (which [html5shiv](https://github.com/aFarkas/html5shiv) will do for you). After running this code, IE8 will behave sanely with all of these elements when manipulating `innerHTML` on document nodes.

However, if the nodes aren't attached to the document, it decides to be sad and not work, and we specifically detach nodes from the document whenever we set `innerHTML` on them: [ReactComponentBrowserEnvironment.js#L131-L145](https://github.com/facebook/react/blob/master/src/browser/ReactComponentBrowserEnvironment.js#L131-L145).

jQuery gets around this by creating the HTML5 elements again in a document fragment before inserting them into the document: https://github.com/jquery/jquery/blob/1.x-master/src/manipulation.js#L18-L30. We should probably do something similar.
